advance-building.rb

```diff
@@ -0,0 +1,27 @@
+def bubble_sort array
+  sorted_array = []
+  while (array.length != 0)
+    array.each_index do |index|
+      if (index != array.length-1 && array[index] > array[index+1])
+        array[index], array[index+1] = array[index+1], array[index]
+      end
+    end
+    sorted_array.insert(0, array.pop)
+  end
+  sorted_array
+end
+
+def bubble_sort_by array
+  n = array.length
+  sorted_array = []
+  while (n != 0)
+    (n-1).times do |i|
+      diff = yield(array[i],array[i+1])
+      if (diff > 0)
+        array[i], array[i+1] = array[i+1], array[i]
+      end
+    end
+    sorted_array.insert(0, array.pop)
+  end
+  sorted_array
+end
```

Comment:
**[REQUIRED]**
You need to make your bubble_sort_by to accept a block just like it is told in the OdinProject [assignment](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks):
![Screenshot from 2019-10-16 16-51-18](https://user-images.githubusercontent.com/30977845/66953575-31be5180-f035-11e9-9195-37f89aab2a80.png)

- Remember to raise an error if `bubble_sort_by` is called without passing a block.